.buildkite/release_pipeline.yaml

```diff
@@ -158,6 +158,10 @@ steps:
 
   - wait
 
+  - label: ":docker: Upload Containers to Docker Hub"
+    command: .buildkite/scripts/dockerhub_upload.sh
+    agents:
+      queue: habitat-release
```

Comment:
In the course of a normal release (not a "fake" one for testing), will the fact that this image is pushed out to DockerHub with its "real" tags cause any issues? I'm thinking of what happens if we push something out, then discover that it's buggy, but by then, someone out in the real world has already consumed it.

Could we have a "dev" repository to push the image to initially, then once we've approved the release and promoted Habitat packages to the stable channel, and published releases in Bintray, we could re-upload the image to a "real" repository? This would essentially be like creating an "unstable channel" in DockerHub.

(For the record, I think we should also do something like this for our images in Bintray)

Reply:
We currently have a release tag of something like `0.60.0-rc` right? If so, I'd say we could use that as the tag for rollback and latest should always point at stable. For example:

```
habitat/default-studio:latest -> habitat/default-studio:0.59.0
habitat/default-studio:0.60.0-rc
habitat/default-studio:0.60.0-dev
habitat/default-studio:0.59.0
```

Reply:
I don't think we've currently got anything like that for Docker images, though that'd also probably work. I don't have super strong feelings exactly which way we separate things, just that we _do_ separate things.

Reply:
The nice thing about doing it with tags over a repo is you won't have to rebuild the image which could potentially introduce drift

Reply:
I don't think you need to rebuild the image, just re-tag it and push.